pkg/schedule/schedulers/balance_range.go

```diff
@@ -167,6 +227,332 @@ func newBalanceRangeScheduler(opController *operator.Controller, conf *balanceRa
 	return s
 }
 
+// Schedule schedules the balance key range operator.
+func (s *balanceRangeScheduler) Schedule(cluster sche.SchedulerCluster, _ bool) ([]*operator.Operator, []plan.Plan) {
+	balanceRangeCounter.Inc()
+	job := s.conf.peek()
+	if job == nil {
+		balanceRangeNoJobCounter.Inc()
+		return nil, nil
+	}
+
+	opInfluence := s.OpController.GetOpInfluence(cluster.GetBasicCluster(), operator.WithRangeOption(job.Ranges))
+	plan, err := s.prepare(cluster, opInfluence, job)
```

Comment:
For every time we schedule, it will scan the range from the beginning?

Reply:
yes, the info on the regions and stores in the given ranges may be updated.

Reply:
Will it affect the other requests like get region or heartbeat if the range is large?

Reply:
yes, the operator of scanning regions acquires Rlock which shares with the region heartbeat.  Maybe we can reuse the previous distribution by interval duration, not every scheduler.